docs/source/daml/reference/interfaces.rst

```diff
@@ -160,32 +152,31 @@ For context, a simple template definition:
    :start-after: -- TEMPLATE_HEADER_BEGIN
    :end-before: -- TEMPLATE_HEADER_END
 
-Interface Instances
--------------------
+``Interface Instance`` clauses
+------------------------------
 
 .. literalinclude:: ../code-snippets-dev/Interfaces.daml
    :language: daml
-   :start-after: -- TEMPLATE_IMPLEMENTS_BEGIN
-   :end-before: -- TEMPLATE_IMPLEMENTS_END
+   :start-after: -- INTERFACE_INSTANCE_BEGIN
+   :end-before: -- INTERFACE_INSTANCE_END
 
-- To make a template implement an interface, an ``interface instance`` clause is added to the body of the template.
+- To make a template an instance of an interface, an ``interface instance`` clause must be defined in either the template declaration or the interface declaration.
+- Either the template or the interface of the clause must match the enclosing declaration. In other words, a template ``T`` declaration can only contain ``interface instance`` clauses where the template is ``T``, and an interface ``I`` declaration can only contain ``interface instance`` clauses where the interface is ``I``.
+- The rules for an ``interface instance`` clause are the same whether the enclosing declaration is a template or an interface.
 - The clause must start with the keywords ``interface instance``, followed by the name of the interface, then the keyword ``for`` and the name of the template, and finally the keyword ``where``, which introduces a block where **all** the methods of the interface must be implemented.
+- Within the clause, there's an implicit local binding ``this`` referring to a contract of the template type. The template parameters of this contract are also in scope.
 - The special ``view`` method must be implemented with the same return type as the interface's view type.
-- Methods can be defined using the same syntax as for top level functions, including pattern matches and guards (e.g. ``method3``).
-
-..TODO(MA): explain interface instances on interfaces
-  https://github.com/digital-asset/daml/issues/14047
+- Method implementations can be defined using the same syntax as for top level functions, including pattern matches and guards (e.g. ``method3``).
```

Comment:
I think it would make sense to specify the scope and purpose of defining an interface instance on the interface itself and introduce the term "retroactive" to establish shared terminology with the readers. To the best of my understanding there's nothing preventing some to define a non-retroactive instance on the interface, but that's probably more of a reason to give guidance on how to follow the intent of the feature.

Reply:
> I think it would make sense to specify the scope and purpose of defining an interface instance on the interface itself and introduce the term "retroactive" to establish shared terminology with the readers. 

That's a good point. I avoided this because I'm not sure what's the best way to introduce it. It would also demand an explanation of the security implications. I'll try to write something and ping you again, if that's okay.

> To the best of my understanding there's nothing preventing some to define a non-retroactive instance on the interface, 

Right, that could only happen if the template and the interface are in the same module, …

> but that's probably more of a reason to give guidance on how to follow the intent of the feature.

… and then we should encourage having the instance on the template.